### PR TITLE
Introduce commit

### PR DESCRIPTION
**Pull request by gigbat**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -37,18 +39,50 @@
...
+        <dependency>
+            <groupId>javax.servlet.jsp.jstl</groupId>
+            <artifactId>javax.servlet.jsp.jstl-api</artifactId>
+            <version>1.2.1</version>
+        </dependency>
+        <dependency>
+            <groupId>taglibs</groupId>
+            <artifactId>standard</artifactId>
+            <version>1.1.2</version>
+        </dependency>
```
Comment (Resci) [inline pom.xml]: You use JSP or JSTL?

File: src/main/java/spring/com/controller/UserController.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import spring.com.service.impl.UserMapperServiceImpl;
+
+@RestController
+@RequestMapping("/user")
+public class UserController {
+    private static final AnnotationConfigApplicationContext context =
+            new AnnotationConfigApplicationContext(AppConfig.class);
+    private static final UserService userService = context.getBean(UserService.class);
+
+    @RequestMapping(value = "/{userId}", method = RequestMethod.GET)
```
Comment (Resci) [inline src/main/java/spring/com/controller/UserController.java]: [Checklist](https://mate-academy.github.io/jv-program-fulltime/04_spring/checklist/hw_03.html) p.4

File: src/main/java/spring/com/controller/UserController.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        User user = null;
+        if (optionalUser.isPresent()) {
+            user = optionalUser.get();
+            return UserMapperServiceImpl.dtoBoxing(user);
+        }
+        throw new RuntimeException("User " + user + " isn't exist");
+    }
+
+    @GetMapping(value = "/inject")
+    public void addUser(@RequestParam String name, @RequestParam String surname,
```
Comment (Resci) [inline src/main/java/spring/com/controller/UserController.java]: I think, inject not same as add

File: src/main/java/spring/com/controller/UserController.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+                        @RequestParam String email) {
+        User user = new User(name, surname, email);
+        userService.add(user);
+    }
+
+    @GetMapping
+    public List<UserResponseDto> getAll() {
```
Comment (Resci) [inline src/main/java/spring/com/controller/UserController.java:44]: Checklist p.9

**Review by Resci — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (gigbat) [reply pom.xml]: done

File: src/main/java/spring/com/controller/UserController.java (hunk 3 above)
Comment (gigbat) [reply src/main/java/spring/com/controller/UserController.java]: done

File: src/main/java/spring/com/controller/UserController.java (hunk 2 above)
Comment (gigbat) [reply src/main/java/spring/com/controller/UserController.java]: done

File: src/main/java/spring/com/controller/UserController.java (hunk 4 above)
Comment (gigbat) [reply src/main/java/spring/com/controller/UserController.java:44]: done

**Review by Resci — APPROVED** [review]

File: src/main/java/spring/com/controller/UserController.java [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+import spring.com.model.User;
+import spring.com.service.UserService;
+import spring.com.service.impl.UserMapperServiceImpl;
+
+@RestController
+@RequestMapping("/user")
+public class UserController {
+    private static final AnnotationConfigApplicationContext context =
+            new AnnotationConfigApplicationContext(AppConfig.class);
+    private static final UserService userService = context.getBean(UserService.class);
```
Comment (Sofasmile) [inline src/main/java/spring/com/controller/UserController.java]: https://mate-academy.github.io/jv-program-fulltime/04_spring/checklist/hw_03.html

File: src/main/java/spring/com/dao/impl/UserDaoImpl.java [hunk 6]
```diff
@@ -48,4 +49,14 @@ public void add(User user) {
             throw new RuntimeException("Can't get all users", e);
         }
     }
+
+    @Override
+    public Optional<User> get(Long id) {
+        try (Session session = sessionFactory.openSession()) {
+            return session.createQuery("from User where id = :id", User.class)
```
Comment (Sofasmile) [inline src/main/java/spring/com/dao/impl/UserDaoImpl.java]: https://mate-academy.github.io/jv-program-fulltime/04_spring/checklist/hw_03.html

File: src/main/java/spring/com/dto/UserResponseDto.java [hunk 7]
```diff
@@ -0,0 +1,31 @@
+package spring.com.dto;
+
+public class UserResponseDto {
+    private String name;
```
Comment (Sofasmile) [inline src/main/java/spring/com/dto/UserResponseDto.java:5]: let's add id fielsd here

File: src/main/java/spring/com/service/UserService.java [hunk 8]
```diff
@@ -1,10 +1,13 @@
...
 import java.util.List;
+import java.util.Optional;
 import spring.com.model.User;
 
 public interface UserService {
     void add(User user);
 
     List<User> listUsers();
+
+    Optional<User> get(Long id);
```
Comment (Sofasmile) [inline src/main/java/spring/com/service/UserService.java]:
```suggestion
    User get(Long id);
```

File: src/main/java/spring/com/service/impl/UserMapperServiceImpl.java [hunk 9]
```diff
@@ -0,0 +1,24 @@
+package spring.com.service.impl;
+
+import org.springframework.stereotype.Service;
+import spring.com.dto.UserResponseDto;
+import spring.com.model.User;
+
+@Service
```
Comment (Sofasmile) [inline src/main/java/spring/com/service/impl/UserMapperServiceImpl.java]:
```suggestion
@Component
```

File: src/main/java/spring/com/service/impl/UserMapperServiceImpl.java [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+public class UserMapperServiceImpl {
```
Comment (Sofasmile) [inline src/main/java/spring/com/service/impl/UserMapperServiceImpl.java]:
```suggestion
public class UserMapper {
```

File: src/main/java/spring/com/service/impl/UserMapperServiceImpl.java [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+    public static UserResponseDto dtoBoxing(User user) {
```
Comment (Sofasmile) [inline src/main/java/spring/com/service/impl/UserMapperServiceImpl.java]: why is this method static?

File: src/main/java/spring/com/service/impl/UserMapperServiceImpl.java [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+        UserResponseDto userResponseDto = new UserResponseDto();
+        userResponseDto.setName(user.getName());
+        userResponseDto.setSurname(user.getSurname());
+        userResponseDto.setEmail(user.getEmail());
+        return userResponseDto;
+    }
+
+    public static User dtoUnboxing(UserResponseDto userResponseDto) {
```
Comment (Sofasmile) [inline src/main/java/spring/com/service/impl/UserMapperServiceImpl.java]: where do we use this method? Let's remove it)

File: src/main/java/spring/com/controller/UserController.java [hunk 13]
```diff
@@ -0,0 +1,55 @@
...
+
+    @GetMapping(value = "/{userId}")
+    public UserResponseDto getUser(@PathVariable Long userId) {
+        Optional<User> optionalUser = userService.get(userId);
+        User user = null;
+        if (optionalUser.isPresent()) {
+            user = optionalUser.get();
+            return UserMapperServiceImpl.dtoBoxing(user);
+        }
+        throw new RuntimeException("User " + user + " isn't exist");
```
Comment (Sofasmile) [inline src/main/java/spring/com/controller/UserController.java]: there are a lot of redundant code here
let's remove all checking for Optional and call `get()` or `orElseThrow()` method at the service layer

**Review by Sofasmile — CHANGES_REQUESTED** [review]

File: src/main/java/spring/com/controller/UserController.java (hunk 5 above)
Comment (gigbat) [reply src/main/java/spring/com/controller/UserController.java]: done

File: src/main/java/spring/com/dao/impl/UserDaoImpl.java (hunk 6 above)
Comment (gigbat) [reply src/main/java/spring/com/dao/impl/UserDaoImpl.java]: done

File: src/main/java/spring/com/dto/UserResponseDto.java (hunk 7 above)
Comment (gigbat) [reply src/main/java/spring/com/dto/UserResponseDto.java:5]: done

File: src/main/java/spring/com/service/UserService.java (hunk 8 above)
Comment (gigbat) [reply src/main/java/spring/com/service/UserService.java]: done

File: src/main/java/spring/com/service/impl/UserMapperServiceImpl.java (hunk 10 above)
Comment (gigbat) [reply src/main/java/spring/com/service/impl/UserMapperServiceImpl.java]: done

File: src/main/java/spring/com/service/impl/UserMapperServiceImpl.java (hunk 11 above)
Comment (gigbat) [reply src/main/java/spring/com/service/impl/UserMapperServiceImpl.java]: i thought that these would look good, but i fixed these

File: src/main/java/spring/com/service/impl/UserMapperServiceImpl.java (hunk 12 above)
Comment (gigbat) [reply src/main/java/spring/com/service/impl/UserMapperServiceImpl.java]: done

File: src/main/java/spring/com/controller/UserController.java (hunk 13 above)
Comment (gigbat) [reply src/main/java/spring/com/controller/UserController.java]: done

File: src/main/java/spring/com/service/impl/UserMapperServiceImpl.java (hunk 9 above)
Comment (gigbat) [reply src/main/java/spring/com/service/impl/UserMapperServiceImpl.java]: done

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**